src/sentry/notifications/helpers.py

```diff
@@ -32,20 +36,28 @@
 def _get_notification_setting_default(
     provider: ExternalProviders,
     type: NotificationSettingTypes,
+    recipient: Team | User | None = None,
 ) -> NotificationSettingOptionValues:
     """
     In order to increase engagement, we automatically opt users into receiving
     Slack notifications if they install Slack and link their identity.
     Approval notifications always default to Slack being on.
     """
+    from sentry.models import User
+
+    if isinstance(recipient, User) and features.has(
+        "users:notification-slack-automatic", None, actor=recipient
+    ):
```

Comment:
It's a bit weird that we pass `None` here and not the recipient.

Reply:
@wedamija Agreed, maybe I should just not use a feature flag here. Thoughts?

Reply:
I think the main thing is whether you have a use for `UserFeature` in the future. If you do, then maybe it's worth trying to make this less clunky. If not, it might be better to just use a setting.

Reply:
@wedamija Good question. I do imagine there is a use case for other notification features that would be user and not organization-specific. So probably worth the investment.